cis194/week6/daniel-deng/src/Exercise07.hs

```diff
@@ -0,0 +1,20 @@
+{-# OPTIONS_GHC -fno-warn-missing-methods #-}
+
+module CIS194.Homework06.Exercise07 (fib4) where
+
+data Mat2x2 a = Mat2x2 a a a a
+
+instance Num a => Num (Mat2x2 a) where
+  Mat2x2 a11 a12 a21 a22 * Mat2x2 b11 b12 b21 b22
```

Comment:
@shineli1984 I find `Mat2x2` clearer, is it violating the Haskell naming convention by putting a digit in the middle?

Reply:
No. It's clearer imo too